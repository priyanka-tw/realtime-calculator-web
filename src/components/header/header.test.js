import React from 'react';
import {render} from '@testing-library/react';
import 'jest-styled-components';
import WS from "jest-websocket-mock";
import Header from "./header";
import Context from "../../services/global-context-provider/context";

describe('Header', () => {
    let server, client;

    beforeEach(async () => {
        server = new WS("ws://localhost:8080/ws", {jsonProtocol: true});
        client = new WebSocket("ws://localhost:8080/ws");
    });

    afterEach(() => {
        WS.clean();
    });
    it('should render the component', () => {
        const {container} = render(<Header/>);

        expect(container).toMatchSnapshot();
    });

    it('should render additional data only when user is logged in', async () => {
        const {container} = render(<Context.Provider value={{username: 'test-username', socket: client}}>
            <Header/></Context.Provider>);
        await server.connected;

        expect(container).toMatchSnapshot();
    });
});
