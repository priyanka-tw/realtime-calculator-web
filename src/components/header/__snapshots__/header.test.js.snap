// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render the component 1`] = `
.c1 {
  radius-border: 5px;
  height: 50px;
  margin-right: 10px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-self: start;
  -ms-flex-item-align: start;
  align-self: start;
  min-width: 100%;
}

.c2 {
  font-size: 20px;
  font-family: verdana;
}

<div>
  <div
    class="c0"
  >
    <img
      class="c1"
      src="calculator.svg"
    />
    <h2
      class="c2"
    >
      calculator
    </h2>
  </div>
</div>
`;
